### PR TITLE
Update documentation: imagePath option, examples, readme, etc.

### PR DESCRIPTION
**Pull request by PeterTheOne**
- Add `imagePath` option to all examples and `examples.html`.
- Add example to README.
- Add more doc to `reference.html`.

### CODE REVIEW

**Comment by PeterTheOne** [comment]
@markmcd Please merge this and close all issues and Pull Requests about `imagePath`.

File: README.md [hunk 1]
```diff
@@ -8,6 +8,43 @@ A Google Maps JavaScript API v3 library to create and manage per-zoom-level clus
...
+              mapTypeId: google.maps.MapTypeId.ROADMAP
+            });
+
+            var markers = [];
+            var marker = new google.maps.Marker({
+                position: new google.maps.LatLng(51.5074, 0.1278)
+            });
+            markers.push(marker);
+
+            var options = {
```
Comment (markmcd) [inline README.md:39]: This is the default now, but I think it's still useful to demonstrate how a user could override with their own images. Could you add a comment to that effect?

File: examples/advanced_example.html [hunk 2]
```diff
@@ -148,7 +148,8 @@
         markerClusterer = new MarkerClusterer(map, markers, {
           maxZoom: zoom,
           gridSize: size,
-          styles: styles[style]
+          styles: styles[style],
+          imagePath: '../images/m'
```
Comment (markmcd) [inline examples/advanced_example.html:152]: These are no longer needed (here and the two files below).

**Comment by markmcd** [comment]
This looks great - thanks for the contribution!

File: README.md (hunk 1 above)
Comment (PeterTheOne) [inline README.md:39]: sure, I will tomorrow.

File: examples/advanced_example.html (hunk 2 above)
Comment (PeterTheOne) [inline examples/advanced_example.html:152]: My intention was to make the `imagePath` option explicit in the examples. If someone copies the example he/she will hopefully notice this line and adapt it to their setup. But I can scrap it if you don't agree..

Comment (markmcd) [inline examples/advanced_example.html:152]: Good idea. Leave it in.

File: README.md (hunk 1 above)
Comment (PeterTheOne) [inline README.md:39]: @markmcd I've made a change in 6d7db4e, should be ready to merge now.

**Comment by markmcd** [comment]
Thanks again for the contribution.

**Comment by PeterTheOne** [comment]
Thanks for the merge! :)

**Pull request merged**